server/src/main/java/org/opensearch/cluster/metadata/IndexMetadata.java

```diff
@@ -278,6 +278,17 @@ public Iterator<Setting<?>> settings() {
         Property.Final
     );
 
+    public static final String SETTING_REMOTE_STORE = "index.remote_store";
+    /**
+     * Used to specify if the index data should be persisted in the remote store.
+     */
+    public static final Setting<Boolean> INDEX_REMOTE_STORE_SETTING = Setting.boolSetting(
+        SETTING_REMOTE_STORE,
+        false,
+        Property.IndexScope,
+        Property.Final
+    );
```

Comment:
Wondering if we should have separate controls for remote segment store and remote translog store

Reply:
Agree, we may need to provide a granular control. Once we finalize these settings, we can either add them or split this one. Let me add it as an issue in the [Meta Issue](https://github.com/opensearch-project/OpenSearch/issues/2992).

Reply:
Or I will keep updating the parent issue of this PR: https://github.com/opensearch-project/OpenSearch/issues/3141.